### PR TITLE
feat: create UI components steps indicator

### LINKED ISSUE #29: Create UI components `steps indicator`
## Description

- The task is to create a shared UI component for `steps indicator`.

## Criteria

- [ ] create a `steps indicator` component

## Notes

image

## Assignment 

|Role|Assignee |
|---|---|
|`Developer`|@Akikaze1119 |
|`Reviewer`|@KaiKoide |

### PR DESCRIPTION
**Pull request by Akikaze1119**
## Overview
The task is to create a shared UI component for `steps indicator`.

## Changes
Created `steps indicator` component based on [figma](https://www.figma.com/file/5tNJDDfqC0HnHbWygzEXiL/Tascurator?type=design&node-id=0-1&mode=design&t=SvzFH1RdR9OWCQRn-0)

## Screenshots or videos
▼Figma
Screenshot 2024-04-09 at 4 49 07 PM

▼Code
https://github.com/Tascurator/tascurator-frontend/assets/132441773/7ce8df02-e98a-4420-8c11-84dc2ec22c69

### About bar style
I used hard code for setting the height of the horizontal bar.
- Figma: 2px
- Code: h-[0.125rem](=2px)
*Note: The minimum height of the Tailwind class is `h-1 (4px)`, which is too long for the bar.

### About the name
Since 'stepper' seems to be more common, both the file name and the component name are set to 'stepper'.
[Reference](https://mui.com/material-ui/react-stepper/)

### About the confirmation code
The following code can be used for review.
```
'use client';
import { useState } from 'react';

import { Stepper } from '@/components/ui/stepper';
import { Button } from '@/components/ui/button';

const INITIAL_STEP = 1;
const MAX_STEPS = 4;

export default function Home() {
  const [currentStep, setCurrentStep] = useState(INITIAL_STEP);

  return (
    <>
      <main className="flex min-h-screen flex-col items-center p-6">
        <Stepper currentStep={currentStep} maxSteps={MAX_STEPS} />

        <div className="flex mt-10 gap-4">
          <Button
            size="sm"
            onClick={() => setCurrentStep((prev) => prev - 1)}
            disabled={currentStep === INITIAL_STEP}
          >
            Previous
          </Button>
          <Button
            size="sm"
            onClick={() => setCurrentStep((prev) => prev + 1)}
            disabled={currentStep === MAX_STEPS}
          >
            Next
          </Button>
        </div>
      </main>
    </>
  );
}
```

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [ ] Code readability and simplicity
- [ ] Follows best practices and coding standards
- [ ] Understandable and maintainable code

### CODE REVIEW

File: src/components/ui/stepper.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+import { cn } from '@/lib/utils';
+
+interface IStep {
+  currentStep: number;
+  maxSteps: number;
+}
+
+const Stepper: React.FC<IStep> = ({ currentStep, maxSteps }: IStep) => {
```
Comment (KaiKoide) [inline src/components/ui/stepper.tsx:39]: Could you add TSDoc so the other team members could apply this component? The way to write TSDoc can be confirmed on Scrapbox.

File: src/components/ui/stepper.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  const steps: number[] = Array.from(
+    { length: maxSteps },
+    (_, index) => index + 1,
+  );
+
+  const width = `${(100 / (maxSteps - 1)) * (currentStep - 1)}%`;
+
+  return (
+    <div className="w-full max-w-2xl mx-auto px-1 pb-10">
+      <div className="flex justify-between relative before:bg-slate-300 before:absolute before:h-[0.125rem] before:top-1/2 before:transform-y-1/2 before:w-full before:left-0">
```
Comment (KaiKoide) [inline src/components/ui/stepper.tsx:49]: The font-weight is 500 so could you add `font-medium`?
image

**Review by KaiKoide — CHANGES_REQUESTED** [review]

**Comment by Akikaze1119** [comment]
@KaiKoide 
Thank you for your review!
I modified codes.

**Review by KaiKoide — APPROVED** [review]
Thank you for the correction! LGTM 👍

**Pull request merged**